tiled/database/orm.py

```diff
@@ -154,16 +154,15 @@ class Role(Timestamped, Base):
 class APIKey(Timestamped, Base):
     __tablename__ = "api_keys"
 
+    # Store the first_eight characters of the hex-encoded secret.
+    # The key holder can use this to identity the key.
+    # We do not store the full secret, only its sha256-hashed value.
+    # A primary key on (first_eight, hashed_secret) enables
+    # fast lookups.
+    first_eight = Column(Unicode(8), primary_key=True, index=True, nullable=False)
```

Comment:
The full hashed value will be unique...8 bytes of the 32? Very low chance of collision, but would this be a case for an incrementing integer for that primary key?

Reply:
The primary key is on `(first_eight, hashed_secret)` so I think this is only a collision if _both_ columns collide.

Likewise with the `Identity` table where the key is `(provider, identity)`. Naturally there are many collisions on `provider` and there may be some collisions on `identity`, but that is OK.

Reply:
Oh cool...I missed that it was a compound primary. Move along, nothing to see here.